### PR TITLE
Remove playback <path> SDF param in Dome

### PR DESCRIPTION
**Pull request by mabelzhang**
Address #293 for Dome

- Remove `<path>` SDF parameter for `LogPlayback`
- Change server-injected `<path>` to `<playback_path>` to disambiguate between the two, similar to what's already done for `<record_path>`
- Update tests accordingly

### CODE REVIEW

File: test/integration/log_system.cc [hunk 1]
```diff
@@ -1594,8 +1569,8 @@ TEST_F(LogSystemTest, LogResources)
   // Recorded models should exist
   EXPECT_GT(entryCount(recordPath), 2);
   EXPECT_TRUE(common::exists(common::joinPaths(recordPath, homeFake,
-      ".ignition", "fuel", "fuel.ignitionrobotics.org", "openrobotics",
-      "models", "x2 config 1")));
+      ".ignition", "fuel", "fuel.ignitionrobotics.org", "OpenRobotics",
```
Comment (JShep1) [inline test/integration/log_system.cc:1572]: Looks like these lines and the following lines on 1607-1608 introduce a regression.  Moving them  back to lower case fixes the test case, is there any reason you uppercased these?

**Review by JShep1 — COMMENTED** [review]
Looks good, just have a comment on the test regression

File: test/integration/log_system.cc (hunk 1 above)
Comment (mabelzhang) [reply test/integration/log_system.cc]: Ohh I must have outdated fuel stuff locally. Changed back in 2c042bc

**Review by JShep1 — APPROVED** [review]
LGTM pending CI

**Comment by JShep1** [comment]
@osrf-jenkins run tests please

**Comment by JShep1** [comment]
Not sure why tests are failing, possibly flaky tests, re-running CI

**Comment by mabelzhang** [comment]
All green except ServerRepeat/SceneBroadcasterTest.State/0, which is flaky: https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/5269/testReport/(root)/ServerRepeat_SceneBroadcasterTest/State_0/history/

Merging.

**Pull request merged**